### DISCUSSION TITLE
downloading attachments

### DISCUSSION THREAD

**Discussion by davetannenbaum** [joelekstrom/fastmate#47] [Q&A]
Hi Joel. First off, thank you for making Fastmate — it's an essential part of my work flow.

When downloading an attachment, Fastmate kicks me over to my default web browser (Safari) to download the file. Is there any way to perform a direct download from within Fastmate?

**Comment by joelekstrom** [comment]
Hi, thanks for using Fastmate! Glad that you enjoy it.

It's not currently possible in the release version. I know @mdbraber has done some work on this, check #34. 

If you want it now you could try his fork. I've been meaning to take a look at merging it and making sure there are progress bars etc, but the inspiration hasn't gotten to me yet, and I can't promise if/when that might happen 😬

**Reply by davetannenbaum** [reply]
Sounds good, thanks @joelekstrom

**Reply by mdbraber** [reply]
Still want to take some time integrating this in the main branch from Joel. My implementation works 90% of the time, although it doesn't support renaming currently existing files, so watch out with that. Also hitting the download button on inlined PDFs also doesn't seem to work yet.

**Comment by mdbraber** [comment]
@davetannenbaum keep an eye out for this PR https://github.com/joelekstrom/fastmate/pull/82 and the next release

**Comment by joelekstrom** [comment]
@davetannenbaum This is now implemented in v1.9.0!
